forbidigo/forbidigo_test.go

```diff
@@ -134,24 +151,86 @@ func ExampleFoo() {
 }`)
 		assert.NotEmpty(t, issues)
 	})
+
+	t.Run("import renames not detected by simple pattern", func(t *testing.T) {
+		linter, _ := NewLinter([]string{`fmt\.Printf`}, OptionExcludeGodocExamples(false))
+		issues := parseFile(t, linter, "file.go", `
+package bar
+
+import fmt2 "fmt"
+
+func ExampleFoo() {
+	fmt2.Printf("here i am")
+}`)
+		assert.Empty(t, issues)
+	})
+
+	t.Run("import renames detected by package pattern", func(t *testing.T) {
+		linter, _ := NewLinter([]string{`(?P<pkg>fmt)\.Printf`}, OptionExcludeGodocExamples(false))
+		expectIssues(t, linter, `
+package bar
+
+import fmt2 "fmt"
+
+func ExampleFoo() {
+	fmt2.Printf("here i am")
+}`, "use of `fmt2.Printf` forbidden by pattern `(?P<pkg>fmt)\\.Printf` at testing.go:7:2")
+	})
+
 }
 
+// sourcePath matches "at /tmp/TestForbiddenIdentifiersdisplays_custom_messages4260088387/001/testing.go".
+var sourcePath = regexp.MustCompile(`at .*/([[:alnum:]]+.go)`)
+
 func expectIssues(t *testing.T, linter *Linter, contents string, issues ...string) {
 	actualIssues := parseFile(t, linter, "testing.go", contents)
 	actualIssueStrs := make([]string, 0, len(actualIssues))
 	for _, i := range actualIssues {
-		actualIssueStrs = append(actualIssueStrs, i.String())
+		str := i.String()
+		str = sourcePath.ReplaceAllString(str, "at $1")
+		actualIssueStrs = append(actualIssueStrs, str)
 	}
 	assert.ElementsMatch(t, issues, actualIssueStrs)
 }
 
 func parseFile(t *testing.T, linter *Linter, fileName, contents string) []Issue {
-	fset := token.NewFileSet()
-	expr, err := parser.ParseFile(fset, fileName, contents, parser.ParseComments)
+	// We can use packages.Load if we put a single file into a separate
+	// directory and parse it with Go modules of. We have to be in that
+	// directory to use "." as pattern, parsing it via the absolute path
+	// from the forbidigo project doesn't work ("cannot import absolute
+	// path").
+	tmpDir := t.TempDir()
```

Comment:
I think we should probably have a call to `t.Cleanup(...)` that removes the directory after the test.

Reply:
`t.TempDir` does that automatically.